pkg/csi/service/fcd/controller_test.go

```diff
@@ -0,0 +1,644 @@
+	log.Println(AttributeFirstClassDiskParentName, ":", respCreate.Volume.VolumeContext[AttributeFirstClassDiskParentName])
+
+	if !strings.EqualFold("test", volName) {
+		t.Errorf("[CREATE] Name of FCD does not match test != %s", volName)
+	}
+
+	//list
+	respList, err := c.ListVolumes(ctx, &csi.ListVolumesRequest{})
+	if err != nil {
+		t.Errorf("ListVolumes failed: %v", err)
+	} else {
+		if len(respList.Entries) != 1 {
+			t.Error("There should only be a single FCD present")
+		} else {
+			name := respList.Entries[0].Volume.VolumeContext[AttributeFirstClassDiskName]
+			if !strings.EqualFold("test", name) {
+				t.Errorf("[LIST] Name of FCD does not match test != %s", name)
+			}
+		}
+	}
+
+	//publish
+	reqPub := &csi.ControllerPublishVolumeRequest{
+		VolumeId: volID,
+		NodeId:   vmName,
+	}
+	respPub, err := c.ControllerPublishVolume(ctx, reqPub)
+	if err != nil {
+		t.Errorf("ControllerPublishVolume failed: %v", err)
+	} else {
+		pubCon := respPub.GetPublishContext()
+
+		name := pubCon[AttributeFirstClassDiskName]
+		if !strings.EqualFold("test", name) {
+			t.Errorf("[PUB] Name of FCD does not match test != %s", name)
+		}
+	}
+
+	//unpublish
+	reqUnpub := &csi.ControllerUnpublishVolumeRequest{
+		VolumeId: volID,
+		NodeId:   vmName,
+	}
+	_, err = c.ControllerUnpublishVolume(ctx, reqUnpub)
+	if err != nil {
+		t.Errorf("ControllerUnpublishVolume failed: %v", err)
+	}
+
+	//delete
+	reqDelete := &csi.DeleteVolumeRequest{
+		VolumeId: volID,
+	}
+	_, err = c.DeleteVolume(ctx, reqDelete)
+	if err != nil {
+		t.Errorf("DeleteVolume failed: %v", err)
+	}
+}
+
+func TestListBoundaries(t *testing.T) {
+	config, cleanup := configFromEnvOrSim(false)
+	defer cleanup()
+
+	connMgr := cm.NewConnectionManager(&config, nil)
+	defer connMgr.Logout()
+
+	c := &controller{
+		cfg:     &config,
+		connMgr: connMgr,
+	}
+
+	//context
+	ctx := context.Background()
+
+	// Get a simulator DS
+	myds := simulator.Map.Any("Datastore").(*simulator.Datastore)
+
+	err := connMgr.Connect(ctx, config.Global.VCenterIP)
+	if err != nil {
+		t.Errorf("Failed to Connect to vSphere: %s", err)
+	}
+
+	//create
+	params := make(map[string]string, 0)
+	params[AttributeFirstClassDiskParentType] = string(vclib.TypeDatastore)
+	params[AttributeFirstClassDiskParentName] = myds.Name
+
+	for i := 1; i <= 11; i++ {
+		volName := fmt.Sprintf("test%d", i)
+		reqCreate := &csi.CreateVolumeRequest{
+			Name: volName,
+			CapacityRange: &csi.CapacityRange{
+				RequiredBytes: int64(i) * GbInBytes,
+			},
+			Parameters: params,
+		}
+
+		respCreate, err := c.CreateVolume(ctx, reqCreate)
+		if err != nil {
+			t.Fatalf("CreateVolume failed: %v", err)
+		}
+
+		volNameRet := respCreate.Volume.VolumeContext[AttributeFirstClassDiskName]
+		if !strings.EqualFold(volName, volNameRet) {
+			t.Errorf("[CREATE] Name of FCD does not match %s != %s", volName, volNameRet)
+		}
+	}
+
+	// get first 10
+	resp, err := c.ListVolumes(context.Background(), &csi.ListVolumesRequest{MaxEntries: 10})
```

Comment:
You have a `ctx` var in scope that could be used in place of  `context.Background()` here and the ones below.

Reply:
Copy and past error. Will update.